server/src/test/java/io/crate/types/TypeCompatibilityTest.java

```diff
@@ -39,8 +39,8 @@ public void test_get_common_type_for_text_with_len_limit_return_text_with_highes
 
     @Test
     public void test_numeric_highest_precision_wins() {
-        assertCommonType(NumericType.INSTANCE, new NumericType(2, 1), new NumericType(2, 1));
-        assertCommonType(new NumericType(2, 1), NumericType.INSTANCE, new NumericType(2, 1));
+        assertCommonType(NumericType.INSTANCE, new NumericType(2, 1), NumericType.INSTANCE);
+        assertCommonType(new NumericType(2, 1), NumericType.INSTANCE, NumericType.INSTANCE);
```

Comment:
I think `null` scale represents the largest possible scale, https://github.com/crate/crate/blob/86594bf0774288794cf30347c9229d30fc2e589d/server/src/main/java/io/crate/types/NumericType.java#L207-L208

Reply:
cc @romseygeek do you remember why we went with more specific having higher precedence?

Looks like the tests originated in https://github.com/crate/crate/commit/5e25b88d53928bc763b1293475cc1c00549ecdb7 

I preserved it in https://github.com/crate/crate/commit/5d555cebbf21114fea0e24cf5a0fded88474dbbb but it initially also confused me because it's different from string (where no length=unbound -> higher precedence)

Reply:
What do you think about letting numerics with any precisions and scales to have the same precedence? As you can notice in this PR, a numeric literal comparision causes a predecence check but if you think about it, any numerics must be allowed to be compared.

Reply:
If you're adding something with a precision of 2 and something with a precision of 3, then I'd say that the result should have a precision of 2, mathematically speaking?  Precision of 2 means 'this value, plus or minus 0.005', precision of 3 means 'this value, plus or minus 0.0005', so giving the result to a precision of 3 is saying things about the accuracy of the result which aren't warranted by the inputs.

String length is different I would argue because it's not about precision.

Reply:
For numeric types and BigDecimals, `precision` represents the number of siginificant digits and `scale` represents the number of digits to the right of the decimal point. The issue that I am stuck on is whether `new NumericType(null, null)` has higher or lower precedence than `new NumericType(2,1)`. My understanding is that `new NumericType(null, null)` has the higher precedence(null = unbounded) such that the tests need to be updated.

Also what would you say about the precedence of these two, `new NumericType(5,2)` and `new NumericType(4,3)`? I think it is undefined in this case.